torchgeo/datasets/pastis.py

```diff
@@ -0,0 +1,410 @@
+        assert set(folds) <= set(range(6))
+        assert bands in ["s1a", "s1d", "s2"]
+        assert mode in ["semantic", "instance"]
+        self.root = root
+        self.folds = folds
+        self.bands = bands
+        self.mode = mode
+        self.transforms = transforms
+        self.download = download
+        self.checksum = checksum
+        self._verify()
+        self.files = self._load_files()
+
+        colors = []
+        for i in range(len(self.cmap)):
+            colors.append(
+                (
+                    self.cmap[i][0] / 255.0,
+                    self.cmap[i][1] / 255.0,
+                    self.cmap[i][2] / 255.0,
+                )
+            )
+        self._cmap = ListedColormap(colors)
+
+    def __getitem__(self, index: int) -> dict[str, Tensor]:
+        """Return an index within the dataset.
+
+        Args:
+            index: index to return
+
+        Returns:
+            data and label at that index
+        """
+        image = self._load_image(index)
+        if self.mode == "semantic":
+            mask = self._load_semantic_targets(index)
+            sample = {"image": image, "mask": mask}
+        elif self.mode == "instance":
+            mask, boxes, labels = self._load_instance_targets(index)
+            sample = {"image": image, "mask": mask, "boxes": boxes, "label": labels}
+
+        if self.transforms is not None:
+            sample = self.transforms(sample)
+
+        return sample
+
+    def __len__(self) -> int:
+        """Return the number of data points in the dataset.
+
+        Returns:
+            length of the dataset
+        """
+        return len(self.idxs)
+
+    def _load_image(self, index: int) -> Tensor:
+        """Load a single time-series.
+
+        Args:
+            index: index to return
+
+        Returns:
+            the time-series
+        """
+        path = self.files[index][self.bands]
+        array = np.load(path)
+
+        tensor = torch.from_numpy(array)
+        return tensor
+
+    def _load_semantic_targets(self, index: int) -> Tensor:
+        """Load the target mask for a single image.
+
+        Args:
+            index: index to return
+
+        Returns:
+            the target mask
+        """
+        # See https://github.com/VSainteuf/pastis-benchmark/blob/main/code/dataloader.py#L201 # noqa: E501
+        # even though the mask file is 3 bands, we just select the first band
+        array = np.load(self.files[index]["semantic"])[0].astype(np.uint8)
+        tensor = torch.from_numpy(array).long()
+        return tensor
+
+    def _load_instance_targets(self, index: int) -> tuple[Tensor, Tensor, Tensor]:
+        """Load the instance segmentation targets for a single sample.
+
+        Args:
+            index: index to return
+
+        Returns:
+            the instance segmentation mask, box, and label for each instance
+        """
+        mask_array = np.load(self.files[index]["semantic"])[0].astype(np.uint8)
```

Comment:
Wonder if we could somehow call `_load_semantic_targets` to avoid code duplication here